pkg/clustermesh/kvstoremesh/kvstoremesh.go

```diff
@@ -80,6 +86,31 @@ func newKVStoreMesh(lc cell.Lifecycle, params params) *KVStoreMesh {
 	return &km
 }
 
+type SyncWaiterParams struct {
+	cell.In
+
+	KVStoreMesh *KVStoreMesh
+	SyncState   health.SyncState
+}
+
+func RegisterSyncWaiter(lc cell.Lifecycle, p SyncWaiterParams) {
+	lc.Append(cell.Hook{
+		OnStart: func(ctx cell.HookContext) error {
```

Comment:
Blocking a start hook will prevent every other start hook from being executed. Additionally, there's a timeout causing the shutdown of the entire hive if they don't terminate on time (IIRC something around 5 minutes).

Reply:
I unblocked the start hook by spawning a new context. This is probably not ideal, but using a start hook is pretty convenient for making sure it runs after the common.ClusterMesh hook completes and the configs have been read.